project_euler/problem_188/sol1.py

```diff
@@ -0,0 +1,53 @@
+"""
+The hyperexponentiation of a number
+Problem 188
+
+The hyperexponentiation or tetration of a number a by a positive integer b,
+denoted by a↑↑b or b^a, is recursively defined by:
+
+a↑↑1 = a,
+a↑↑(k+1) = a(a↑↑k).
+
+Thus we have e.g. 3↑↑2 = 3^3 = 27, hence 3↑↑3 = 3^27 = 7625597484987 and
+3↑↑4 is roughly 103.6383346400240996*10^12.
+
+Find the last 8 digits of 1777↑↑1855.
+"""
+
+
+def solution(a=1777, k=1855, digits=8):
```

Comment:
Type hints: https://docs.python.org/3/library/typing.html

Please use descriptive names instead of one letter names like `a` and `k`

Reply:
But it's literally "a" and "k" in the problem itself, should I add that as a comment then? like 

> "a" in the problem is called "base" here

or similar?